rmarkdowns/Week_4.Rmd

```diff
@@ -257,10 +260,261 @@ For your challenges this week, create the necessary code chunks below. You can c
+
+dim(inat_obs_sf)
+#opened the file and it has the correct file with Mamaki on it 
+
+```
+
+```{r}
+
+#Taxon ID = 121022 NONI (Native Hawaiian Medicinal Plant)
+#Place ID = 11 Hawaii, USA
+inat_obs_df1 <- get_inat_obs(taxon_id = 121022, #taxon id from URL
+                            place_id = 11, #place id from URL
+                            quality = "research", #specifies research grade only
+                            geo = TRUE, #Specifies that we want geocoordinates
+                            maxresults = 100) #Limits results
+
+save(inat_obs_df1, file = "hawaii_noni_inat.Rdata")
+
+inat_obs_sf1 <-  inat_obs_df1 %>% 
+  select(longitude, latitude, datetime, common_name, scientific_name, image_url, user_login) %>% 
+  st_as_sf(coords=c("longitude", "latitude"), crs=4326)
+
+# FYI note to myself what this code means (since I kept forgetting) - crs = 4326 ; set the map projection2 to a common projection standard such as WGS84 via the argument crs = 4326
+
+dim(inat_obs_sf1)
+#opened the file and it has the correct file with Mamaki on it 
+
+```
+
+
+
 **Challenge 2: Create a static map using the `ggplot` function of your target area and target genus/species.**
 
+```{r}
+#For Mamaki
+Mamaki <- ggplot() +
+  geom_point(data = inat_obs_df, aes(x = longitude, y = latitude, color = common_name, text = common_name)) 
+Mamaki
+```
+
+
+```{r}
+#For Noni
+Noni <- ggplot() +
+  geom_point(data = inat_obs_df1, aes(x = longitude, y = latitude, color = common_name, text = common_name)) 
+Noni
+```
+
+```{r}
+#Combining 2 ggplot for Mamaki & Noni
+
+# Problem that I am still unable to resolve it yet - How to input the legend on the side to identify which one is Mamaki vs Noni.  Tried to input legend but it didn't work. Tried to input color=common_name but the result stated common_name was not found
+
+# Shall I give up and move on to the next challenge? ;) 
+
+Combined_Plot <- ggplot() + 
+geom_point(data=inat_obs_df, aes(x = longitude, y = latitude), color="darkgray", fill ='#8dd3c7', size=3, shape = 21) +
+geom_point(data=inat_obs_df1, aes(x = longitude, y = latitude), color= "darkgray", fill = '#bebada', size=3, shape = 21) +
+  theme_minimal() +
+    geom_text(color = "white", size = 20) +
+  labs(title = "Medicinal Plants in Hawaii", 
+       subtitle = "Mamaki (Piper Methysticum) and Noni (Morinda Citrifolia)") 
+Combined_Plot
+
+```
+
+```{r}
+#I decided to take a snack break, but I couldn't stop thinking about the issue at hand.  Eventually, I discovered a more efficient method to combine the two plots with the legend on the side. 
+
+# Lesson Learned: As a programmer, we often get fixated on the code and stuck on a problem.  Instead of forcing ourselves to resolve it immediately, it's beneficial to take a break and clear our minds. Often, the solution will present itself naturally :) 
```

Comment:
Love this! I was about to comment this solution when I saw that you solved it on your own!